lib/nostrum/shard/session.ex

```diff
@@ -22,50 +27,74 @@ defmodule Nostrum.Shard.Session do
       end
 
     payload = Payload.status_update_payload(idle_since, game, stream, status, afk, type)
-    WebSockex.cast(pid, {:status_update, payload})
+    GenServer.cast(pid, {:status_update, payload})
   end
 
   def request_guild_members(pid, guild_id, limit \\ 0) do
     payload = Payload.request_members_payload(guild_id, limit)
-    WebSockex.cast(pid, {:request_guild_members, payload})
+    GenServer.cast(pid, {:request_guild_members, payload})
   end
 
   def start_link([gateway, shard_num]) do
+    GenServer.start_link(__MODULE__, [gateway, shard_num])
```

Comment:
We should add the option of setting the `fullsweep_after` start option here. It's a `spawn_opt` as you can see [here](https://hexdocs.pm/elixir/Process.html#t:spawn_opt/0) that users can choose to use so their bot uses less memory.

You can see how we implemented it before [here](https://github.com/Kraigie/nostrum/blob/ffb73b62d68a55b6adf8298a257a42734a6b4368/lib/nostrum/shard/session.ex#L48). Even then it wasn't doing anything back then, it will work now.